src/mcts/search.cc

```diff
@@ -452,10 +457,29 @@ EdgeAndNode Search::GetBestChildWithTemperature(Node* parent,
 
   assert(parent->GetChildrenVisits() > 0);
   std::vector<float> cumulative_sums;
+  std::vector<EdgeAndNode> edges;
+  auto edge_iterator = parent->Edges();
   float sum = 0.0;
   const float n_parent = parent->GetN();
 
-  for (auto edge : parent->Edges()) {
+  // Load edges to vector.
+  for (auto& edge : edge_iterator) edges.push_back(edge);
+
+  // Filter edges that do not have enough visits.
+  edges.erase(std::remove_if(edges.begin(), edges.end(),
+                             [&](auto edge) {
+                               return edge.GetN() <=
+                                      static_cast<unsigned int>(
+                                          kMinimumTemperatureVisits);
+                             }),
+              edges.end());
+
+  // Avoids crash if all edges have been filtered.
+  if (edges.empty()) {
+    for (auto& edge : edge_iterator) edges.push_back(edge);
+  }
```

Comment:
I tried initializing the vector from the iterator at one point, and after failing, tried using a combination of std::copy and std::copy_if. I failed that, too. I'm not really a C++ programmer, but I spent about two hours trying to find a cleaner way of doing things, and got nowhere. I'm super sorry for not being able to write better code than that.

Can you elaborate on this? 

> You can filter directly at after root limit check. Then later instead of checking root limit again and idx--, you can random access at index idx instead.

Reply:
After limit root check (still inside the loop) push into filterededges if <= n, and to edges else. If edges is empty, set edges = filterededges. Then at the second limit root check, you can entirely remove the for loop, and use edges[idx] instead

Reply:
That's genius. I will do that.

Reply:
Okay, in an attempt to implement that, I've come to the conclusion that it requires maintaining an additional sum vector. Is there a way of avoiding that?

Reply:
make sum after choosing which edge vector to use